conda_build/metadata.py

```diff
@@ -846,7 +859,9 @@ def _get_dependencies_from_environment(env_name_or_path):
     return {"requirements": {"build": bootstrap_requirements}}
 
 
-def toposort(output_metadata_map):
+def toposort(
+    output_metadata_map: dict[frozendict, OutputTuple],
+) -> list[OutputTuple]:
```

Comment:
(The current deprecation framework doesn't handle typing changes -- IDK if we care much about it here, but just wanted to note that this not too different from other signature changes, deprecation wise.)

Reply:
I've added back the original functions and instead implemented the fixes in new functions (`toposort` → `_toposort_outputs`, `check_circular_dependencies` → `_check_circular_dependencies`)